pkg/storage/formattedstore/formattedstore.go

```diff
@@ -137,17 +154,24 @@ type formatStore struct {
 	cacheStore            storage.Store
 	formatter             Formatter
 	skipIteratorFiltering bool
+	batch                 *Batch
 }
 
 func (s *formatStore) Put(k string, v []byte) error {
-	formattedValue, err := s.formatter.FormatPair(k, v)
-	if err != nil {
-		return fmt.Errorf(failFormat, err)
-	}
+	if s.batch != nil {
+		if err := s.batch.Put(k, v); err != nil {
+			return err
+		}
+	} else {
+		formattedValue, err := s.formatter.FormatPair(k, v)
```

Comment:
Should you put formatted value into batch? This way formatting error will be caught upfront.

Reply:
If there is not chance for user error than it is ok to delay encryption